lib/postgrex/protocol.ex

```diff
@@ -62,13 +63,14 @@ defmodule Postgrex.Protocol do
           {:ok, state}
           | {:error, Postgrex.Error.t() | %DBConnection.ConnectionError{}}
   def connect(opts) do
-    {host, port} = host_and_port(opts)
+    {host, port, remaining_endpoints_to_attempt} = host_and_port_params(opts)
```

Comment:
I feel you should always return a list here and then you traverse the list in `connect_endpoints`. This will avoid mistakes like the one in line 94 below where the types_key has the wrong host and port (they need to be computed later now).

Reply:
I agree with your suggested change (that the function should always return a list) but not `types_key`. The way I see it, if more than one endpoint is provided, only one endpoint will be a primary instance and the others will be read replicas. In that case, if I'm not mistaken, types should be shared across all instances. Therefore, it's enough to keep the same value for `types_key`, no matter which endpoint we actually connect to. This will avoid doing unnecessary bootstraps. What do you think?

Reply:
I would prefer to always make it conditional to the host we connected. What if a node went down and then we tried one of the fallbacks and the fallbacks has a more recent version due to a new deployment?

Reply:
True. I made the change